process/block/baseProcess.go

```diff
@@ -454,6 +465,11 @@ func (bp *baseProcessor) requestHeadersIfMissing(sortedHdrs []data.HeaderHandler
 			return process.ErrNilRequestHeaderHandlerByNonce
 		}
 
+		isHeaderOutOfRange := nonce > lastNotarizedHdrNonce+allowedSize
```

Comment:
sort missingNonces then you can request only the allowedSize

Reply:
They are already sorted and the allowedSize is used to request only the maximum occupancy percentage allowed to be used from the full pool capacity, for the received data which are not needed in the near future.